services/debian/debian.service.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const Joi = require('joi')
+const { latest, renderVersionBadge } = require('../version')
+const { BaseJsonService, NotFound } = require('..')
+
+const schema = Joi.array()
+  .items(
+    Joi.object().pattern(
+      /./,
+      Joi.object()
+        .pattern(/./, Joi.object().pattern(/./, Joi.object()))
+        .required()
+    ) // Optional, missing means not found
+  )
+  .required()
+
+const defaultDistribution = 'stable'
+
+module.exports = class Debian extends BaseJsonService {
+  static get category() {
+    return 'version'
+  }
+
+  static get route() {
+    return {
+      base: 'debian/v',
+      pattern: ':distribution?/:packageName',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Debian package',
+        namedParams: { distribution: 'unstable', packageName: 'apt' },
+        staticPreview: renderVersionBadge({ version: '1.8.0' }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'debian' }
+  }
+
+  async handle({ distribution = defaultDistribution, packageName }) {
+    const data = await this._requestJson({
+      schema,
+      url: `https://api.ftp-master.debian.org/madison?f=json&s=${distribution}&package=${packageName}`,
+    })
+    if (!data.length) {
+      throw new NotFound()
+    }
+    // Distribution can change compared to request, for example requesting
+    // "stretch" may map to "stable" in results, so can't use value from
+    // request as is. Just take the first one instead.
+    for (const resultdist of Object.keys(data[0][packageName])) {
+      const versions = Object.keys(data[0][packageName][resultdist])
+      return renderVersionBadge({ version: latest(versions) })
+    }
+    throw new NotFound()
```

Comment:
Is this line reachable?

Reply:
It is, if a package is returned without any versions. I suppose that shouldn't happen, but if it does, "not found" seems appropriate behavior to me.